include/rocksdb/cache.h

```diff
@@ -34,6 +34,12 @@ namespace rocksdb {
 
 class Cache;
 
+#ifdef ROCKSDB_DEFAULT_TO_ADAPTIVE_MUTEX
```

Comment:
This is compiler flag when building RocksDB. include/rocksdb is the directory for header files when users use to build. We can't assume they specify the same compiler flag when they build their project. So using ROCKSDB_DEFAULT_TO_ADAPTIVE_MUTEX here is not appropriated. We can hide the declaration in code outside include/rocksdb if you want. Or, we can just make it default true, and explain the API change in HISTORY.md.

Reply:
Thanks, makes sense! Will move the definition to a .cc file.